### PR TITLE
FEATURE: BB-1411 manage pivot subtotals via submenu

### PR DESCRIPTION
**Pull request by yanlipnican**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3834/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3089/ : SUCCESS in 14m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3835/ : SUCCESS in 53s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3090/ : SUCCESS in 5m 45s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3838/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3094/ : SUCCESS in 10m 10s

File: src/components/visualizations/table/totals/utils.ts [hunk 1]
```diff
@@ -18,15 +18,39 @@ export const isNativeTotal = (total: AFM.ITotalItem) => {
...
 
-export const getNativeTotals = (totals: AFM.ITotalItem[]): AFM.INativeTotalItem[] => {
+export const getAttributeDimension = (
+    attributeIdentifier: string,
+    resultSpec: AFM.IResultSpec,
+): AFM.IDimension => {
+    return (
+        resultSpec.dimensions.find(
+            dimension => !!dimension.itemIdentifiers.find(attribute => attribute === attributeIdentifier),
+        ) || null
```
Comment (martinmosko) [inline src/components/visualizations/table/totals/utils.ts]: this returns null when dimension is not found... but then result is use in functions
`isGrandTotal`
`getAllSubtotalsOfTypeFromDimension` which are not expecting null as dimension...

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3843/ : SUCCESS in 44s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3097/ : FAILURE in 9m 48s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-893.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3207/ : SUCCESS in 5m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3955/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1055/ : SUCCESS in 34m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3956/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3208/ : SUCCESS in 5m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3957/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3209/ : SUCCESS in 10m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3958/ : SUCCESS in 9s

File: src/components/core/pivotTable/AggregationsSubMenu.tsx [hunk 2]
```diff
@@ -92,7 +92,11 @@ export default class AggregationsSubMenu extends React.Component<IAggregationsSu
                     <Item checked={isSelected} key={attributeLocalIdentifier}>
                         <div
                             onClick={onClick}
-                            className="gd-aggregation-menu-item-inner s-menu-aggregation-inner"
+                            // TODO: BB-1411 fix this
```
Comment (petrjanu) [inline src/components/core/pivotTable/AggregationsSubMenu.tsx]: TODO

File: src/components/core/PivotTable.tsx [hunk 3]
```diff
@@ -665,42 +665,21 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
...
+        );
 
         this.props.pushData({
             properties: {
                 totals: newColumnTotals,
             },
         });
         this.setState({ columnTotals: newColumnTotals });
+
+        this.setGroupingProvider(this.props.groupRows && this.state.sortedByFirstAttribute);
```
Comment (petrjanu) [inline src/components/core/PivotTable.tsx]: Called twice with same arguments > please move to some function

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: src/components/core/PivotTable.tsx (hunk 3 above)
Comment (yanlipnican) [reply src/components/core/PivotTable.tsx]: Done

File: src/components/core/pivotTable/AggregationsSubMenu.tsx (hunk 2 above)
Comment (yanlipnican) [reply src/components/core/pivotTable/AggregationsSubMenu.tsx]: Done

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-893.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3219/ : SUCCESS in 6m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3978/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1065/ : FAILURE in 26m 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3979/ : SUCCESS in 16s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3220/ : SUCCESS in 3m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3980/ : SUCCESS in 8s

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-893.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3221/ : SUCCESS in 10m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3981/ : SUCCESS in 11s

**Comment by petrjanu** [comment]
LGTM

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1066/ : FAILURE in 29m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3982/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1721/ : SUCCESS in 23m 32s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3222/ : SUCCESS in 4m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/884/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/898/ : FAILURE in 23m 12s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1722/ : SUCCESS in 13m 59s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3224/ : SUCCESS in 3m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/885/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/899/ : SUCCESS in 33m 53s

**Pull request merged**